File: symfony-app/web/src/Entity/Appointment.php
<?php

namespace App\Entity;

use App\Repository\AppointmentRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AppointmentRepository::class)]
class Appointment
{
    public const STATUS_PENDING = 'Pending';
    public const STATUS_CONFIRMED = 'Confirmed';
    public const STATUS_CANCELLED = 'Cancelled';
    public const STATUS_COMPLETED = 'Completed';
    public const STATUS_URGENT = 'URGENT';
    public const STATUS_EXPIRED = 'Expired';

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $scheduledAt = null;

    #[ORM\Column(length: 50)]
    private ?string $status = self::STATUS_PENDING;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $reason = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(nullable: false)]
    private ?User $patient = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(nullable: false)]
    private ?User $doctor = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $department = null;

    #[ORM\OneToOne(mappedBy: 'appointment', cascade: ['persist', 'remove'])]
    private ?Teleconsultation $teleconsultation = null;

    #[ORM\OneToMany(targetEntity: AppointmentRating::class, mappedBy: 'appointment', cascade: ['persist', 'remove'], orphanRemoval: true)]
    private Collection $ratings;

    public function __construct()
    {
        $this->ratings = new ArrayCollection();
    }

    public function getId(): ?int { return $this->id; }

    public function getScheduledAt(): ?\DateTimeInterface { return $this->scheduledAt; }
    public function setScheduledAt(?\DateTimeInterface $scheduledAt): static { $this->scheduledAt = $scheduledAt; return $this; }

    public function getStatus(): ?string { return $this->status; }
    public function setStatus(string $status): static { $this->status = $status; return $this; }

    public function getReason(): ?string { return $this->reason; }
    public function setReason(?string $reason): static { $this->reason = $reason; return $this; }

    public function getPatient(): ?User { return $this->patient; }
    public function setPatient(?User $patient): static { $this->patient = $patient; return $this; }

    public function getDoctor(): ?User { return $this->doctor; }
    public function setDoctor(?User $doctor): static { $this->doctor = $doctor; return $this; }

    public function getDepartment(): ?string { return $this->department; }
    public function setDepartment(?string $department): static { $this->department = $department; return $this; }

    public function getTeleconsultation(): ?Teleconsultation { return $this->teleconsultation; }
    public function setTeleconsultation(?Teleconsultation $teleconsultation): static
    {
        if ($teleconsultation !== null && $teleconsultation->getAppointment() !== $this) {
            $teleconsultation->setAppointment($this);
        }
        $this->teleconsultation = $teleconsultation;
        return $this;
    }

    /**
     * @return Collection<int, AppointmentRating>
     */
    public function getRatings(): Collection
    {
        return $this->ratings;
    }

    public function addRating(AppointmentRating $rating): static
    {
        if (!$this->ratings->contains($rating)) {
            $this->ratings->add($rating);
            $rating->setAppointment($this);
        }
        return $this;
    }

    public function removeRating(AppointmentRating $rating): static
    {
        if ($this->ratings->removeElement($rating)) {
            if ($rating->getAppointment() === $this) {
                $rating->setAppointment(null);
            }
        }
        return $this;
    }
}

File: javafx-app/src/main/java/org/docbook/entities/records/Appointment.java
package org.docbook.entities.records;

import java.time.LocalDateTime;

public class Appointment {
    public static final String STATUS_PENDING = "Pending";
    public static final String STATUS_CONFIRMED = "Confirmed";
    public static final String STATUS_CANCELLED = "Cancelled";
    public static final String STATUS_COMPLETED = "Completed";
    public static final String STATUS_EXPIRED = "Expired";

    private Integer id;
    private LocalDateTime scheduledAt;
    private String department;
    private String doctor;
    private String message;
    private String status;
    private int patientId;
    private int doctorId;
    private Teleconsultation teleconsultation;

    public Appointment() {
        this.status = STATUS_PENDING;
    }

    public Appointment(Integer id, LocalDateTime scheduledAt, String department, String doctor, String message, String status) {
        this.id = id;
        this.scheduledAt = scheduledAt;
        this.department = department;
        this.doctor = doctor;
        this.message = message;
        this.status = status != null ? status : STATUS_PENDING;
    }

    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }

    public LocalDateTime getScheduledAt() { return scheduledAt; }
    public void setScheduledAt(LocalDateTime scheduledAt) { this.scheduledAt = scheduledAt; }

    public String getDepartment() { return department; }
    public void setDepartment(String department) { this.department = department; }

    public String getDoctor() { return doctor; }
    public void setDoctor(String doctor) { this.doctor = doctor; }

    public String getMessage() { return message; }
    public void setMessage(String message) { this.message = message; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public int getPatientId() { return patientId; }
    public void setPatientId(int patientId) { this.patientId = patientId; }

    public int getDoctorId() { return doctorId; }
    public void setDoctorId(int doctorId) { this.doctorId = doctorId; }

    public Teleconsultation getTeleconsultation() { return teleconsultation; }
    public void setTeleconsultation(Teleconsultation teleconsultation) { this.teleconsultation = teleconsultation; }

    @Override
    public String toString() {
        return "Appointment{id=" + id + ", scheduledAt=" + scheduledAt + ", department='" + department + "', doctor='" + doctor + "', status='" + status + "'}";
    }
}